azurerm/automation_variable.go

```diff
@@ -0,0 +1,218 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/automation/mgmt/2015-10-31/automation"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/validate"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func parseAzureRmAutomationVariableValue(resource string, input *string) (interface{}, error) {
+	if input == nil {
+		if resource != "azurerm_automation_null_variable" {
+			return nil, fmt.Errorf("Expected value \"nil\" to be %q, actual type is \"azurerm_automation_null_variable\"", resource)
+		}
+		return nil, nil
+	}
+
+	var value interface{}
+	var err error
+	actualResource := "Unknown"
+	datePattern := regexp.MustCompile(`"\\/Date\((-?[0-9]+)\)\\/"`)
+	matches := datePattern.FindStringSubmatch(*input)
+
+	if len(matches) == 2 && matches[0] == *input {
+		if ticks, err := strconv.ParseInt(matches[1], 10, 64); err == nil {
+			value = time.Unix(ticks/1000, ticks%1000*1000000).In(time.UTC)
+			actualResource = "azurerm_automation_datetime_variable"
+		}
+	} else if value, err = strconv.Unquote(*input); err == nil {
+		actualResource = "azurerm_automation_string_variable"
+	} else if value, err = strconv.ParseBool(*input); err == nil {
+		actualResource = "azurerm_automation_bool_variable"
+	} else if value, err = strconv.ParseInt(*input, 10, 32); err == nil {
+		value = int32(value.(int64))
+		actualResource = "azurerm_automation_int_variable"
+	}
+
+	if actualResource != resource {
+		return nil, fmt.Errorf("Expected value %q to be %q, actual type is %q", *input, resource, actualResource)
+	}
+	return value, nil
+}
+
+func AutomationVariableCommonSchemaFrom(s map[string]*schema.Schema) map[string]*schema.Schema {
+
+	varSchema := map[string]*schema.Schema{
+		"resource_group_name": resourceGroupNameSchema(),
+
+		"name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.NoEmptyStrings,
+		},
+
+		"automation_account_name": {
+			Type:         schema.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validate.NoEmptyStrings,
+		},
+
+		"description": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+
+		"encrypted": {
+			Type:     schema.TypeBool,
+			Optional: true,
+			Default:  false,
+		},
+	}
+	return azure.MergeSchema(s, varSchema)
```

Comment:
as discussed IRL can we remove the null resource and pass in the `type` and `validation` for the `value` field here

Reply:
Fixed